integration-tests/src/test/java/io/opentelemetry/OtlpExporterIntegrationTest.java

```diff
@@ -0,0 +1,355 @@
+import io.opentelemetry.proto.metrics.v1.ResourceMetrics;
+import io.opentelemetry.proto.metrics.v1.Sum;
+import io.opentelemetry.proto.trace.v1.InstrumentationLibrarySpans;
+import io.opentelemetry.proto.trace.v1.ResourceSpans;
+import io.opentelemetry.proto.trace.v1.Span.Link;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.export.IntervalMetricReader;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.IdGenerator;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.SimpleSpanProcessor;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * Integration test to verify OTLP gRPC and HTTP exporters work. Verifies integration with the
+ * OpenTelemetry Collector. The Collector is configured to accept OTLP data over gRPC and HTTP and
+ * to export data over gRPC to a server running in process, allowing assertions to be made against
+ * the data.
+ */
+@Testcontainers(disabledWithoutDocker = true)
+class OtlpExporterIntegrationTest {
+
+  private static final String COLLECTOR_IMAGE = "otel/opentelemetry-collector:0.33.0";
+  private static final Integer COLLECTOR_OTLP_GRPC_PORT = 4317;
+  private static final Integer COLLECTOR_OTLP_HTTP_PORT = 4318;
+  private static final Integer COLLECTOR_HEALTH_CHECK_PORT = 13133;
+  private static final Integer IN_PROCESS_OTLP_GRPC_PORT = 4319;
+  private static final Resource RESOURCE =
+      Resource.getDefault().toBuilder()
+          .put(ResourceAttributes.SERVICE_NAME, "integration test")
+          .build();
+  private static final Logger LOGGER =
+      Logger.getLogger(OtlpExporterIntegrationTest.class.getName());
+
+  static {
+    if (DockerClientFactory.instance().isDockerAvailable()) {
+      // If docker is available, expose the port the in-process OTLP gRPC server will run on before
+      // the collector is initialized so the collector can connect to it.
+      exposeHostPorts(IN_PROCESS_OTLP_GRPC_PORT);
```

Comment:
Is this necessary? I don't see us having to do this in, for example, the `JaegerRemoteSamplerIntegrationTest` which is needing to do something similar.

Reply:
oh I see. We're sending OTLP back to this process again from the collector!

Reply:
I guess this will cause the test to fail if 4319 is already used? If that's right, it'd be nice if we could avoid it. If we move to BeforeAll, I guess we'd also be able to reference the server's automatic port (without passing an explicit value to `httpPort`)

Reply:
Yes. Was being a bit lazy I suppose, but best to do everything we can to avoid port conflicts.